maro/simulator/scenarios/finance/sub_engines/stock/stock_business_engine.py

```diff
@@ -0,0 +1,270 @@
+        self._stock_list: list = None
+        self._readers: dict = None
+        self._order_mode = OrderMode.market_order
+        self._trader = None
+        self._item_picker = None
+        self._account: SubAccount = None
+
+        self._action_scope_max = self._config["action_scope"]["max"]
+
+        self._init_reader()
+        self._init_trader(self._config)
+
+        self._pending_orders = []
+        self._canceled_orders = []
+
+        self._init_frame()
+        self._build_stocks()
+
+    @property
+    def finance_type(self):
+        return FinanceType.stock
+
+    @property
+    def account(self):
+        return self._account
+
+    @property
+    def frame(self):
+        return self._frame
+
+    @property
+    def snapshot_list(self):
+        return self._snapshots
+
+    @property
+    def name_mapping(self):
+        return {stock.index: code for code, stock in self._stocks_dict.items()}
+
+    def step(self, tick: int):
+        valid_stocks = []
+
+        for code, _ in self._readers.items():
+            for raw_stock in self._item_picker[code].items(tick):
+                if raw_stock is not None:
+                    # update frame by code
+                    stock: Stock = self._stocks_dict[code]
+                    stock.fill(raw_stock)
+                    if stock.is_valid:
+                        valid_stocks.append(code)
+
+        # append cancel_order event
+        decision_event = DecisionEvent(
+            tick, FinanceType.stock, -2, self.name, self._action_scope, action_type=ActionType.cancel_order
+        )
+        evt = self._event_buffer.gen_cascade_event(tick, DecisionEvent, decision_event)
+        self._event_buffer.insert_event(evt)
+        # append account event
+        decision_event = DecisionEvent(
+            tick, FinanceType.stock, -1, self.name, self._action_scope, action_type=ActionType.transfer
+        )
+        evt = self._event_buffer.gen_cascade_event(tick, DecisionEvent, decision_event)
+        self._event_buffer.insert_event(evt)
+        # append order event
+        for valid_stock in valid_stocks:
+            decision_event = DecisionEvent(
+                tick, FinanceType.stock, valid_stock, self.name,
+                self._action_scope, action_type=ActionType.order
+            )
+            evt = self._event_buffer.gen_cascade_event(tick, DecisionEvent, decision_event)
+
+            self._event_buffer.insert_event(evt)
+
+    def post_step(self, tick: int):
+        # after take snapshot, we need to reset the stock
+        self.frame.take_snapshot(tick)
+
+        for stock in self._stock_list:
+            stock.reset()
+
+    def take_action(self, action: Action, remaining_money: float, tick: int) -> TradeResult:
+        # 1. can trade -> bool
+        # 2. return (stock, sell/busy, stock_price, number, tax)
+        # 3. update stock.account_hold_num
+        ret = TradeResult(self.name, action.item_index, 0, tick, 0, 0, False, False)
```

Comment:
The number of parameters not responds  the the parameter list of ``TradeResult``. Same to other instances of ``TradeResult``

Reply:
done